tests/system/small/test_dataframe_io.py

```diff
@@ -247,23 +249,145 @@ def test_to_pandas_array_struct_correct_result(session):
     )
 
 
-def test_load_json(session):
-    df = session.read_gbq(
-        """SELECT
-        JSON_OBJECT('foo', 10, 'bar', TRUE) AS json_column
-        """
-    )
-
+def test_load_json_w_unboxed_py_value(session):
+    sql = """
+        SELECT 0 AS id, JSON_OBJECT('boolean', True) AS json_col,
+        UNION ALL
+        SELECT 1, JSON_OBJECT('int', 100),
+        UNION ALL
+        SELECT 2, JSON_OBJECT('float', 0.98),
+        UNION ALL
+        SELECT 3, JSON_OBJECT('string', 'hello world'),
+        UNION ALL
+        SELECT 4, JSON_OBJECT('array', [8, 9, 10]),
+        UNION ALL
+        SELECT 5, JSON_OBJECT('null', null),
+        UNION ALL
+        SELECT
+            6,
+            JSON_OBJECT(
+                'dict',
+                JSON_OBJECT(
+                    'int', 1,
+                    'array', [JSON_OBJECT('bar', 'hello'), JSON_OBJECT('foo', 1)]
+                )
+            ),
+    """
+    df = session.read_gbq(sql, index_col="id")
+
+    assert df.dtypes["json_col"] == db_dtypes.JSONDtype()
+    assert isinstance(df["json_col"][0], dict)
+
+    assert df["json_col"][0]["boolean"]
+    assert df["json_col"][1]["int"] == 100
+    assert math.isclose(df["json_col"][2]["float"], 0.98)
+    assert df["json_col"][3]["string"] == "hello world"
+    assert df["json_col"][4]["array"] == [8, 9, 10]
+    assert df["json_col"][5]["null"] is None
+    assert df["json_col"][6]["dict"] == {
+        "int": 1,
+        "array": [{"bar": "hello"}, {"foo": 1}],
+    }
+
+
+def test_load_json_to_pandas_has_correct_result(session):
+    df = session.read_gbq("SELECT JSON_OBJECT('foo', 10, 'bar', TRUE) AS json_col")
+    assert df.dtypes["json_col"] == db_dtypes.JSONDtype()
     result = df.to_pandas()
-    expected = pd.DataFrame(
-        {
-            "json_column": ['{"bar":true,"foo":10}'],
-        },
-        dtype=pd.ArrowDtype(pa.large_string()),
-    )
-    expected.index = expected.index.astype("Int64")
-    pd.testing.assert_series_equal(result.dtypes, expected.dtypes)
-    pd.testing.assert_series_equal(result["json_column"], expected["json_column"])
+
+    # The order of keys within the JSON object shouldn't matter for equality checks.
+    pd_df = pd.DataFrame(
+        {"json_col": [{"bar": True, "foo": 10}]},
+        dtype=db_dtypes.JSONDtype(),
+    )
+    pd_df.index = pd_df.index.astype("Int64")
+    pd.testing.assert_series_equal(result.dtypes, pd_df.dtypes)
+    pd.testing.assert_series_equal(result["json_col"], pd_df["json_col"])
+
+
+def test_load_json_in_struct(session):
```

Comment:
Perhaps this replaces the above test? If so, please add a docstring indicating this is a regression test for issue 381148539.

```suggestion
def test_load_json_in_struct(session):
    """Avoid regressions for internal issue 381148539."""
```